### PR TITLE
feat(myst plugin): add plugin for MyST

### PR DESCRIPTION
**Pull request by nokome**
Currently just initial setup.

### CODE REVIEW

**Comment by rowanc1** [comment]
👀🚀

Excited about this, let me know when to jump in!

**Comment by nokome** [comment]
@rowanc1 You're help would be awesome! I'm going to merge this now and @vijay-prema is going to make a start fleshing it out over the next could of weeks.

File: plugins/myst/package.json [hunk 1]
```diff
@@ -22,6 +23,7 @@
   "dependencies": {
     "@stencila/plugin": "^2.0.0-alpha.1",
     "@stencila/types": "^2.0.0-alpha.25",
+    "mdast": "^3.0.0",
```
Comment (rowanc1) [inline plugins/myst/package.json:26]: We also have `myst-spec` and `myst-spec-ext` which is our staging ground for types. Those two libraries are the main ones that are defining our mdast-extended types.

File: plugins/myst/src/decode/inlines.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    switch (md.type) {
+      case "text":
+        return new Text(md.value);
+      case "emphasis":
+        return new Emphasis(mdsToInlines(md.children));
+      case "strong":
+        return new Strong(mdsToInlines(md.children));
+      case "inlineCode":
+        return new CodeInline(md.value);
+      case "mystRole":
```
Comment (rowanc1) [inline plugins/myst/src/decode/inlines.ts:46]: Depending on how you are implementing the `mystRole`/`mystDirective` is mostly a parsing container, we generally get rid of these by the time we render / translate to other formats. For example here, we remove the `sup` role, and just use the children in a paragraph.

[image](https://mystmd.org/sandbox?tab=ast&myst=ewBzAHUAcAB9AGAAYQBzAGQAZgBgAA%3D%3D)

File: plugins/myst/src/snapshots/marks.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
+{
+  "type": "Article",
+  "content": [
+    {
+      "type": "Paragraph",
```
Comment (rowanc1) [inline plugins/myst/src/snapshots/marks.json:5]: Our types are lower case (and camelCase for things like `inlineCode`), maybe these are stencilas types?

**Review by rowanc1 — COMMENTED** [review]
Nice - looks like you are up to speed with the `myst-parser`, left a few thoughts as I looked through it!

File: plugins/myst/src/decode/inlines.ts (hunk 2 above)
Comment (nokome) [reply plugins/myst/src/decode/inlines.ts:46]: Yep, I think we are on the same page here, this is just a `case` to unwrap the `mystRole`.

File: plugins/myst/src/snapshots/marks.json (hunk 3 above)
Comment (nokome) [reply plugins/myst/src/snapshots/marks.json:5]: Yes, these are snapshots of Stencila Schema types transformed from MyST types.

File: plugins/myst/src/decode/inlines.ts (hunk 2 above)
Comment (rowanc1) [reply plugins/myst/src/decode/inlines.ts:46]: There are a lot of transformations that we run on the AST, the most basic of which [are here](https://github.com/executablebooks/mystmd/blob/main/packages/myst-transforms/src/basic.ts#L18). This one [specifically lifts the roles/directives](https://github.com/executablebooks/mystmd/blob/feat/remote-url/packages/myst-transforms/src/basic.ts#L20).

**Pull request merged**